File: HW9/huan1588/Unique/RedHeadDuck.cpp
#include <iostream>
#include "RedHeadDuck.h"
#include "Quack.h"
#include "FlyWithWings.h"
#include <memory>

RedHeadDuck::RedHeadDuck( ) {
   quackBehavior = std::make_unique<Quack>( );
   flyBehavior = std::make_unique<FlyWithWings>( );
}

void RedHeadDuck::display( ) {
   std::cout << "I’m a real redheaded duck" << std::endl;
}

File: HW14/huan1588/RedHeadDuck.java
public class RedHeadDuck extends Duck {
   public RedHeadDuck( ) {
      quackBehavior = new Quack( );
      flyBehavior = new FlyWithWings( );
      eggBehavior = new LaysEggBroody();
   }

   public void display( ) {
      System.out.println("I’m a real redheaded duck");
   }
}
